edx/edx-enterprise
# [ENT-685] Improve landing page UX for when item is unenrollable.

**Description:** This PR improves the landing page UX for when the course or program is unenrollable. How that's determined is given by the function `is_course_run_enrollable` being run on the course(s) -- see the docstring or the JIRA ticket.

**JIRA:** [ENT-685](https://openedx.atlassian.net/browse/ENT-685)

**Merge deadline:** End of sprint

**Testing instructions:**

1. Configure a course run to have the end date before now, or use any of the other criteria found in the course-run enrollability criteria to make the course be unenrollable.
2. Go to the [course enrollment landing page](https://business.sandbox.edx.org/enterprise/2b5a2956-7746-4fc9-9587-bc887ba45973/course/course-v1:edX+DemoX+Demo_Course/enroll/) and notice that you get a message at the top indicating that the course is unavailable for enrollment at this time, and that track selection & the continue button don't show. You'll see something like this:

![selection_002](https://user-images.githubusercontent.com/10018065/31584857-55884d82-b1d0-11e7-86df-ef009d09413c.jpg)

3. Put this course in a program (doesn't matter if all other courses are enrollable), and go to the [program landing page](https://business.sandbox.edx.org/enterprise/2b5a2956-7746-4fc9-9587-bc887ba45973/program/52ad909b-c57d-4ff1-bab3-999813a2479b/enroll/). Notice that you see the message at the top, and also that the purchase/price text, as well as the "continue" button, are hidden. Make sure that the modals work. You'll see something like this:

![selection_003](https://user-images.githubusercontent.com/10018065/31584974-a7ce4676-b1d2-11e7-8b74-600ea01378b3.jpg)
Continuation:
![selection_004](https://user-images.githubusercontent.com/10018065/31584975-a99d95c4-b1d2-11e7-82f4-c37104bde00e.jpg)

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**
